dht.go

```diff
@@ -95,7 +97,7 @@ func New(ctx context.Context, h host.Host, options ...opts.Option) (*IpfsDHT, er
 // NewDHT creates a new DHT object with the given peer as the 'local' host.
 // IpfsDHT's initialized with this function will respond to DHT requests,
 // whereas IpfsDHT's initialized with NewDHTClient will not.
-func NewDHT(ctx context.Context, h host.Host, dstore ds.Batching) *IpfsDHT {
+func NewDHT(ctx context.Context, h host.Host, dstore ds.Batching) *Node {
```

Comment:
How about we take the plunge and rename this func to `NewNode`? It's more idiomatic in Go, as the package is already named `dht`:

```
node := dht.NewNode()
```

We'd need to introduce a method alias, though.

Reply:
Yes, I concur. I also want to rename NewDHTClient to NewClientNode, or NewPassiveNode.

Reply:
let's not unnecessarily break existing code.

Reply:
An alias would be added for a period for backwards compatibility until the next major version.